### ISSUE TITLE
Invoke the Reconciler Loop

### ISSUE THREAD

**Issue by cnuss** [operator-framework/operator-sdk#741]
## Type of question

**I'm asking for general context for the operator-sdk and working with the Reconciler lifecycle.**

## Question

 - I've implemented the Reconciler method.
 - It has successfully created my resources after I did a `kubectl apply`
 - The reconciler "loop" has finished running, i.e. the cluster is in the proper state and has not set up a requeue.

Is there a way to invoke another `kubectl apply` (and `kubectl apply --force`) type command to kick off the Reconciler from the top?  My manifest file hasn't changed, and I'd prefer not to make a minuscule change it to start the Reconciler from the top.

In layman's terms, the pods I have are perhaps in a bad state and I need to "kick" the Reconciler so it can fix them, but remain idle at all other times.

**What did you see instead? Under which circumstances?**
If I invoke `kubectl apply` I get a message (for example):

```
kubectl apply -f deploy/crds/my_v1alpha1_controllertype_cr.yaml
controllertype.my.domain.baz/example unchanged
```

**Environment**
* operator-sdk version: 0.1.0

https://github.com/operator-framework/operator-sdk/commit/ec5387ceebfe7a33b53fd49711d294d93c0fb264

* Kubernetes version information:

  insert output of `kubectl version` here

* Kubernetes cluster kind: 
```
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.3", GitCommit:"a4529464e4629c21224b3d52edfe0ea91b072862", GitTreeState:"clean", BuildDate:"2018-09-10T11:44:36Z", GoVersion:"go1.11", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:05:37Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by joelanford** [comment]
@cnuss there are a few ways to trigger reconciliation without your custom resource changing.

> In layman's terms, the pods I have are perhaps in a bad state and I need to "kick" the Reconciler so it can fix them, but remain idle at all other times.

In this case, you'd want to setup another watch on your controller that uses the `EnqueueRequestForOwner`  event handler. The default scaffolding for a new controller does this by default:
```go
err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
	IsController: true,
	OwnerType:    &yourapiversion.YourKind{},
})
```

This will trigger your reconciler if a Pod changes that has an owner reference for your custom resource.

The `handler.EventHandler` parameter to watch is an interface, so you can use that in conjuction with one or more predicates in your `Watch` call to implement any logic you'd like for triggering reconciliation.

Lastly, if you just need to periodically refresh you can also use the `RequeueAfter` field in the `reconcile.Result` when you return from your `Reconcile` method.
```go
return reconcile.Result{RequeueAfter: 5 * time.Minute}, nil
```

**Comment by joelanford** [comment]
@cnuss If this doesn't answer your question or if you have follow-ups feel free to re-open!
